src/TerminalFighter/TextArea/TextArea.cpp

```diff
@@ -0,0 +1,24 @@
+#include "TextArea.h"
+
+TextArea::TextArea(std::string t, SDL_Color c, int x, int y, int w, int h, std::string font_path, int font_size)
+    : text_str_(t)
+    , color_(c)
+    , x_pos_(x)
+    , y_pos_(y)
+    , width_(w)
+    , height_(h)
+    , font_(TTF_OpenFont(font_path.c_str(), font_size)) {
+}
+
+TextArea::~TextArea() {}
+
+void TextArea::render(SDL_Renderer& renderer_) {
+    SDL_Surface* surface = TTF_RenderText_Solid(font_, text_str_.c_str(), color_);
+    SDL_Texture* Message = SDL_CreateTextureFromSurface(&renderer_, surface);
+    SDL_Rect Message_rect;
+    Message_rect.x = x_pos_;
+    Message_rect.y = y_pos_;
+    Message_rect.w = width_;
+    Message_rect.h = height_;
+    SDL_RenderCopy(&renderer_, Message, NULL, &Message_rect);
```

Comment:
SDL_Texture is also created in dynamic memory. After SDL_RenderCopying it, you can safely delete it:

`SDL_DestroyTexture(Message);`

Note you have to clean this up all manually when you do SDL_RenderCopy by yourself.

When you use the GraphicsHandler, it will do this all for you:

This is an example of it being used:
https://github.com/JeLLyNinjas/TerminalFighter/blob/5709f99229da6e750ebdf0f829e7eafae1df5abd/src/TerminalFighter/Missile/Missile.cpp#L30

And here is the line where graphics handler does cleanup:
https://github.com/JeLLyNinjas/TerminalFighter/blob/5709f99229da6e750ebdf0f829e7eafae1df5abd/src/Engine/GraphicsHandler/GraphicsHandler.cpp#L73

In hindsight.. there is no comments on Graphics handler yet (in progress). I will make sure to finish this right now lol...